packages/use-optionally-controlled-state/src/index.tsx

```diff
@@ -1,23 +1,31 @@
-import {useState} from 'react';
+import {useState, useCallback, useRef} from 'react';
 import useConstant from 'use-constant';
 
+export type OptionalState<Value> =
+  | {
+      controlledValue?: Value;
+      initialValue: Value;
+      onChange?(value: Value): void;
+    }
+  | {
+      controlledValue: Value;
+      initialValue?: Value;
+      onChange?(value: Value): void;
+    };
+
 /**
  * Enables a component state to be either controlled or uncontrolled.
  */
 export default function useOptionallyControlledState<Value>({
   controlledValue,
   initialValue,
   onChange
-}: {
-  controlledValue?: Value;
-  initialValue?: Value;
-  onChange?(value: Value): void;
-}): [Value | undefined, (value: Value) => void] {
+}: OptionalState<Value>): [Value, (value: Value) => void] {
   const isControlled = controlledValue !== undefined;
   const initialIsControlled = useConstant(() => isControlled);
   const [stateValue, setStateValue] = useState(initialValue);
 
-  if (__DEV__) {
+  if (typeof __DEV__ !== 'undefined' && __DEV__) {
```

Comment:
This is rewritten at build-time, please revert this change.

Reply:
I don't know what is writing `__DEV__`. I would guess it assumes there is a Webpack `DefinePlugin` somewhere replacing the variable.
While this works out of the box in CRA, I think this is prone to bugs and will throw an `__DEV__ is not defined` error. So, I strongly suggest adding this safeguard.

if you still think it's unnecessary, I will remove it :)

Reply:
There are separate dev and prod builds, this variable never makes it into the final bundle: https://unpkg.com/browse/use-optionally-controlled-state@1.1.0/dist/

So yes, please remove the check 🙂

Reply:
ooh, right

Reply:
removed